osu.Game/Rulesets/Ruleset.cs

```diff
@@ -158,7 +158,7 @@ protected Ruleset()
 
         public abstract DifficultyCalculator CreateDifficultyCalculator(WorkingBeatmap beatmap);
 
-        public virtual PerformanceCalculator CreatePerformanceCalculator(WorkingBeatmap beatmap, ScoreInfo score) => null;
+        public virtual PerformanceCalculator CreatePerformanceCalculator(WorkingBeatmap beatmap, DifficultyAttributes attributes, ScoreInfo score) => null;
```

Comment:
Why is this still taking in a `WorkingBeatmap`? It's unused entirely.

Additionally, we will probably want to have something like

```csharp
public PerformanceCalculator CreatePerformanceCalculator(WorkingBeatmap beatmap, ScoreInfo score)
{
    var difficultyCalculator = CreateDifficultyCalculator(beatmap);
    var difficultyAttributes = difficultyCalculator.Calculate(score.Mods);
    return CreatePerformanceCalculator(beatmap, difficultyAttributes, score);
}
```

It doesn't make this change non-breaking for ruleset implementors, but it *does* give time for consumers of this method to adapt. And [there *are* consumers](https://github.com/search?l=C%23&q=CreatePerformanceCalculator&type=Code), [`osu-tools`](https://github.com/ppy/osu-tools) included.

In fact we might want to add that and keep that forever as again I'm not sure whether making consumers have to create a difficulty calculator to be even able to use the performance calculator is good design.

Reply:
looks like I had a brain fart at the moment I committed this 😅